File: app/src/main/java/pl/edu/pb/todoapp/TaskListActivity.java
package pl.edu.pb.todoapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import android.os.Bundle;

public class TaskListActivity extends SingleFragmentActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragment_task_list);

        FragmentManager fragmentManager = getSupportFragmentManager();
        Fragment fragment = createFragment();

        if(fragment == null){
            fragment = new TaskListFragment();
            fragmentManager.beginTransaction()
                    .add(R.id.fragment_container,fragment)
                    .commit();
        }
    }

    @Override
    protected Fragment createFragment() {
        FragmentManager fragmentManager = getSupportFragmentManager();
        return fragmentManager.findFragmentById(R.id.fragment_task_list);
    }
}

File: app/src/main/java/pl/edu/pb/todoapp/TaskListActivity.kt
package pl.edu.pb.todoapp

import android.os.Bundle
import android.os.PersistableBundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class TaskListActivity : SingleFragmentActivity() {


    override fun createFragment(): Fragment {
        return TaskListFragment()
    }

}